Add spring boot acmeair benchmark application

vpa1977 opened the pull request:
#### Changes proposed in this pull request:

 - This pull request adds benchmark application for the chiselled container. It consists of the Spring boot application based on IBM's sample:
   *  https://github.com/vpa1977/acmeair-monolithic-java
   This repository provides a jmeter script that can be used to measure throughput of the application using `jmeter`. 

Next steps:
  - write report with benchmark results and add it to README.md:
       * comparative throughput of official vs chiselled docker
       * startup times
       * container sizes
  -  add more tests to the application and fix some cosmetic issues.
  - run build and test in a chiselled docker, so that it is a part of the pipeline

- [*] I have signed the [CLA](http://www.ubuntu.com/legal/contributors/)?

-----

*Picture of a cool ROCK:*
![image](https://user-images.githubusercontent.com/1978165/220884085-d4854d10-e1f9-48d5-8400-22aaa2176eca.png)

vpa1977 committed:
add build script

vpa1977 committed:
remove petclinic reference

vpa1977 committed:
liberty deployment

vpa1977 committed:
Docker files  for benchmark application

cjdcordeiro commented on benchmark/README.md:
```diff
@@ -0,0 +1,32 @@
+# Acme Air Sample and Benchmark (monolithic simple spring boot version)
+
+This application shows an implementation of a fictitious airline called "Acme Air".  The application was built with some key business requirements: the ability to scale to billions of web API calls per day, the need to develop and deploy the application targeting multiple cloud platforms (including Public, Private and hybrid).  The application can be deployed both on-prem as well as on Cloud plat ...
```
```suggestion
This application shows an implementation of a fictitious airline called "Acme Air".  The application was built with some key business requirements: the ability to scale to billions of web API calls per day, and the need to develop and deploy the application targeting multiple cloud platforms (including Public, Private and hybrid).  The application can be deployed both on-prem as well as on Cloud platforms.
```

cjdcordeiro commented on benchmark/README.md:
```diff
@@ -0,0 +1,32 @@
+
+# Building & Running
+
+Run `./build.sh` to  clone and build acmeair project.
```
```suggestion
Run `./build.sh` to clone and build acmeair project.
```

cjdcordeiro commented on benchmark/build.sh:
```diff
@@ -0,0 +1,21 @@
+PROJECT=acmeair-monolithic-java
+ACMEAIR_REPO=https://github.com/vpa1977/${PROJECT}
+
+git clone $ACMEAIR_REPO || \
```
could you please create a `tag` in the upstream repo (or at least pin a commit) so that your upstream HEAD can move on without affecting this benchmark

cjdcordeiro commented on benchmark/acmeair-ctomcat/Dockerfile:
```diff
@@ -0,0 +1,38 @@
+ENV PATH=$CATALINA_HOME/bin:$PATH
+WORKDIR $CATALINA_HOME
+
+ENV TOMCAT_NATIVE_LIBDIR $CATALINA_HOME/native-jni-lib
```
mega nitpick, but could you please use the same syntax? (either "=" or " ") :stuck_out_tongue:

cjdcordeiro reviewed (approved):
Uh nice. thanks. this repo is building up quite nicely :)

I'm approving with just a new nitpicks

vpa1977 committed:
Update benchmark/README.md

Co-authored-by: Cristovao Cordeiro <cristovao.cordeiro@canonical.com>
Signed-off-by: Vladimir Petko <vladimir.petko@canonical.com>

vpa1977 committed:
Update benchmark/README.md

Co-authored-by: Cristovao Cordeiro <cristovao.cordeiro@canonical.com>
Signed-off-by: Vladimir Petko <vladimir.petko@canonical.com>

vpa1977 committed:
use uniform env syntax in ctomcat Dockerfile

vpa1977 committed:
add eol in build.sh

vpa1977 committed:
add eol

vpa1977 committed:
Fix exposed ports - no changes for the sample, but allows to connect to mongo

vpa1977 committed:
add missing load database step

vpa1977 committed:
build without packing in tomcat jars

vpa1977 committed:
do not run app twice - disable startup deployment

vpa1977 committed:
review comments: use fixed tag and reduce war size

vpa1977 commented:
I've added some other small changes, namely:
  - build app without bundling tomcat (-6 mb compressed)
  - updated readme (was missing database load step)
  - fixed mongo ports (does not matter for sample, but stops you from debugging =) )
  So probably its worth looking at it again.

vpa1977 committed:
build jar file for liberty

vpa1977 committed:
add tmpfs mounts

vpa1977 committed:
remove accidental debug mode

Signed-off-by: Vladimir Petko <vladimir.petko@canonical.com>

vpa1977 committed:
add missing tmp mounts

vpa1977 committed:
add build output to gitignore

vpa1977 committed:
remove commented out line

cjdcordeiro reviewed (approved):
alright, thanks. lgtm ;)
